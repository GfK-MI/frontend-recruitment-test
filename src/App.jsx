import React from 'react';
import Style from './App.scss';

export default () => <div className={Style.something}>Hello World</div>;
