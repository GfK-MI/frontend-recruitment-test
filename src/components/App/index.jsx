import React, {Component} from 'react';

import UserList from '../UserList';
import Style from './styles.scss';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      query: null
    };

    this.onChangeInput = this.onChangeInput.bind(this);
    this.onSubmitForm = this.onSubmitForm.bind(this);
  }

  onChangeInput(e) {
    e.preventDefault();
    this.setState({inputValue: e.target.value});
  }

  onSubmitForm(e) {
    e.preventDefault();
    this.setState(state => ({query: state.inputValue}));
  }

  render() {
    const {query} = this.state;
    return (
      <>
        <form onSubmit={this.onSubmitForm}>
          <input
            className={Style.searchInput}
            type="text"
            placeholder="search github users"
            onChange={this.onChangeInput}
          />
          <button
            className={Style.searchButton}
            type="submit">
              Search
          </button>
        </form>
        {query && <UserList query={query} />}
      </>
    );
  }
}

export default App;
