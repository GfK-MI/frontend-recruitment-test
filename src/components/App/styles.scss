@import '../../stylesheets/colours';
@import '../../stylesheets/spacing';

$searchBaseLineHeight: 36px;
$searchBaseFontSize: 36px * .4;

.searchBase {
  border: 1px solid $inputBorder;
  box-sizing: border-box;
  font-size: $searchBaseFontSize;
  line-height: $searchBaseLineHeight;
  padding: $baseSpacingS;
}

.searchInput {
  width: $baseSpacing * 20;
  @extend .searchBase;
}

.searchButton {
  @extend .searchBase;
  background-color: $inputBorder;
  border-left-color: transparent;
  color: $white;
}
