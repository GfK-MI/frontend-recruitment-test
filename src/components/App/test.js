import React from 'react';

import {shallow} from 'enzyme';
import App from './';

describe('<App />', () => {
  it('should render correctly', () => {
    const wrap = shallow(<App/>);

    expect(wrap).toMatchSnapshot();
  });

  it('should update query', () => {
    const wrap = shallow(<App/>);
    const inputEl = wrap.find('input');
    const formEl = wrap.find('form');

    inputEl.simulate('change', {target: {value: 'abc'}, preventDefault: Function.prototype})
    formEl.simulate('submit', {preventDefault: Function.prototype});

    expect(wrap).toMatchSnapshot();
  });
});
