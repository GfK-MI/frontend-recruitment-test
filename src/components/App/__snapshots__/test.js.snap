// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      className="searchInput"
      onChange={[Function]}
      placeholder="search github users"
      type="text"
    />
    <button
      className="searchButton"
      type="submit"
    >
      Search
    </button>
  </form>
</Fragment>
`;

exports[`<App /> should update query 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      className="searchInput"
      onChange={[Function]}
      placeholder="search github users"
      type="text"
    />
    <button
      className="searchButton"
      type="submit"
    >
      Search
    </button>
  </form>
  <UserList
    query="abc"
  />
</Fragment>
`;
